lib/constructs/interface.js

```diff
@@ -1311,40 +1311,40 @@ class Interface {
   }
 
   generateOffInstanceMethods() {
-    const addOne = (name, args, body) => {
+    const addOne = (name, args, body, { isAsync, isStatic }) => {
       this.str += `
-        ${name}(${formatArgs(args)}) {${body}}
+        ${isStatic ? "static " : ""}${isAsync ? "async " : ""}${name}(${formatArgs(args)}) {${body}}
```

Comment:
Do we actually need the function to be async? I don't see `await` being used anywhere, and I fear that this difference could in fact be observable by the number of ticks it passed before the promise is rejected with a synchronously thrown error.

Reply:
I talk about that in <https://github.com/jsdom/webidl2js/issues/79#issuecomment-617738500>.

Reply:
It's a convenient way to convert exceptions into rejections. But, it sounds like folks are in favor of instead wrapping everything in try/catch, so I'll do that.